tests/tripal_chado/api/TripalChadoMViewsAPITest.php

```diff
@@ -0,0 +1,215 @@
+  }
+  /**
+   * Test re-populating a materialized view.
+   *
+   * @group api
+   */
+  public function test_chado_populate_mview() {
+    // TODO: this is currently a stub for a test function that neds
+    // implementation. For now it returns true to get past unit testing.
+    $this->assertTrue(true);
+  }
+  /**
+   * Test modifications to a materialized view
+   *
+   * @group api
+   */
+  public function test_chado_get_mview_id() {
+    // TODO: this is currently a stub for a test function that neds
+    // implementation. For now it returns true to get past unit testing.
+    $this->assertTrue(true);
+  }
+  /**
+   * Test retrieving names of the materialized views.
+   *
+   * @group api
+   */
+  public function test_chado_get_mview_table_names() {
+    // TODO: this is currently a stub for a test function that neds
+    // implementation. For now it returns true to get past unit testing.
+    $this->assertTrue(true);
+  }
+  /**
+   * Test retrieving all materialized view objects.
+   *
+   * @group api
+   */
+  public function test_chado_get_mviews() {
+    // TODO: this is currently a stub for a test function that neds
+    // implementation. For now it returns true to get past unit testing.
+    $this->assertTrue(true);
+  }
+  
+  /**
+   * Issue 322 reported the problem of re-adding a materialized view after
+   * the actual table had been manually removed outside of Tripal.  The
+   * function reported errors.
+   *
+   * @ticket 322
+   */
+  public function test_re_adding_deleted_mview_issue_322() {
+    
+    // Add the analysis_organism mview.
+    $mview_name = 'analysis_organism_test';
+    $mview_module = $this->example_mviews[$mview_name]['module'];
+    $mview_sql = $this->example_mviews[$mview_name]['sql'];
+    $mview_schema = $this->example_mviews[$mview_name]['schema'];
+    $mview_comment = $this->example_mviews[$mview_name]['comment'];
+    
+    $success = chado_add_mview($mview_name, $mview_module, $mview_schema, $mview_sql, $mview_comment, FALSE);
+    $this->assertTrue($success, "Failed to create materialized view: $mview_name");
+        
+    // Now simulate manual deletion of the table outside of the API.
+    chado_query('DROP TABLE {' . $mview_name . '}');
+    
+    // Make sure the table no longer exists.
+    $this->assertFalse(chado_table_exists($mview_name), 
+      "Failed to manually remove the materialized view, cannot complete the test.");
+    
+    // Now try to read the mview. Previously, the behavior was the the mview
+    // table would not be created because Tripal thinks it's already there.
+    chado_add_mview($mview_name, $mview_module, $mview_schema, $mview_sql, $mview_comment, FALSE);
+    $this->assertTrue($success, "Failed to re-create materialized view: $mview_name");
+    
+    $this->assertTrue(chado_table_exists($mview_name),
+      "Manually removing a materialized views throws off the chado_add_mview function when the mview is re-added. See Issue #322");    
```

Comment:
Maybe  "Failed to re-add mview that was previously deleted manually. See Issue #322"